server/channels/app/post.go

```diff
@@ -2644,3 +2601,86 @@ func (a *App) MoveThread(c request.CTX, postID string, sourceChannelID, channelI
 	c.Logger().Info(msg)
 	return nil
 }
+
+func (a *App) PermanentDeletePost(c request.CTX, postID, deleteByID string) *model.AppError {
+	post, err := a.Srv().Store().Post().GetSingle(sqlstore.RequestContextWithMaster(c), postID, true)
+	if err != nil {
+		return model.NewAppError("DeletePost", "app.post.get.app_error", nil, "", http.StatusBadRequest).Wrap(err)
+	}
+
+	if len(post.FileIds) > 0 {
+		appErr := a.PermanentDeleteFilesByPost(c, post.Id)
+		if appErr != nil {
+			return appErr
+		}
+	}
+
+	err = a.Srv().Store().Post().PermanentDeletePost(c, post.Id)
+	if err != nil {
+		return model.NewAppError("PermanentDeletePost", "app.post.permanent_delete_post.error", nil, "", http.StatusInternalServerError).Wrap(err)
+	}
+
+	appErr := a.AfterPostDeletionCleanUp(c, post, deleteByID)
+	if appErr != nil {
+		return appErr
+	}
+
+	return nil
+}
+
+func (a *App) AfterPostDeletionCleanUp(c request.CTX, post *model.Post, deleteByID string) *model.AppError {
+	channel, appErr := a.GetChannel(c, post.ChannelId)
+	if appErr != nil {
+		return appErr
+	}
+
+	if post.RootId == "" {
+		if appErr := a.DeletePersistentNotification(c, post); appErr != nil {
+			return appErr
+		}
+	}
+
+	postJSON, err := json.Marshal(post)
```

Comment:
We publish a few websockets in different places with the `post.message` contents after deletion, I initially thought this was an issue because posts are sometimes deleted due to security concerns of the message contents. I tried just setting `post.message` to `""` here but `RemoveNotifications` needs the message contents in order to figure out if there are mentions in the message. I was thinking of just removing the message contents for `WebsocketEventPostDeleted` websockets but we can just leave it in for the `WebsocketEventThreadUpdated` websocket. The `WebsocketEventThreadUpdated` websocket only gets published if a thread comment is deleted

Reply:
It could also be a breaking change to `hooks.MessageHasBeenDeleted` if we remove the message contents